okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/ConcurrencyLimitingInterceptor.java

```diff
@@ -52,7 +46,7 @@
  * 429 and 503 response codes are used for backpressure, whilst 200 -> 399 request codes are used for determining
  * new limits and all other codes are not factored in to timings.
  * <p>
- * Concurrency permits are only released when the response body is closed.
+ * Concurrency permits are released when the response is received.
```

Comment:
I think this was discussed in the past and in general would be kinda against the principle of concurrency limiting: the fact that the body is being streamed usually means that the worker thread might have been released, but streaming the body still usually consumes resources (especially if it's just a blocking read that goes from some external resource and writes to the response body).

I think the reason why you are not observing any 429 in general in that situation is that the internal product that I expect you're referring too has request/sec rate limiting implemented, not concurrency limiting. I suspect releasing more of requests here would eventually lead to 503s from the product.